rules/linux/defense_evasion_kthreadd_masquerading.toml

```diff
@@ -0,0 +1,88 @@
+[metadata]
+creation_date = "2024/02/01"
+integration = ["endpoint"]
+maturity = "production"
+min_stack_comments = "New fields added: required_fields, related_integrations, setup"
+min_stack_version = "8.3.0"
+updated_date = "2024/02/01"
+
+[rule]
+author = ["Elastic"]
+description = """
+Monitors for kernel processes with associated process executable fields that are not empty. Unix kernel processes such 
+as kthreadd and kworker typically do not have process.executable fields associated to them. Attackers may attempt to
+hide their malicious programs by masquerading as legitimate kernel processes.  
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.*"]
```

Comment:
is this rule compatible also with other indexes ?

Reply:
++, added endgame & auditd_manager support. Will check others.

Reply:
When adding auditd_manager support, I noticed I **must** supply the `event.dataset` value in the query - this currently does not make sense. Will check with Terrance to see whether we can extend coverage without being forced to do this. Created this Meta for it: https://github.com/elastic/detection-rules/issues/3428.